recipe/meta.yaml

```diff
@@ -24,11 +24,13 @@ requirements:
     - pip
     - setuptools
   run:
+    # See https://pypi.org/pypi/kombu/{{ version }}/json
     - amqp >=5.1.1,<6.0.0
     - typing_extensions  # strictly, py<310
```

Comment:
this also, rather obnoxiously, is pinned very tightly upstream... it reckon it's either worth emulating that (kind ick) or patching the `requirements/defaults.txt` to make it a `>=`, and verifying with `mypy`... tests might be out of the question for this one...

Reply:
I'll add ">=4.12.2". If it causes a problem, then I'd be tempted to change the feedstock to be 3.10+ rather than pollute the ecosystem with more javascript-style exact pins (https://iscinumpy.dev/post/bound-version-constraints/).

Reply:
yeah, preaching to the choir about hard pins.


Some options:
- patch out the `==` to be a `>=`, and verifying with e.g. `mypy`
  - the patch will break every time upstream changes, but that's _good_ in this case.
- dynamically patch with dumb python script
  - this is probably also fine
- use a variant in `conda-build-config.yaml` to generate multiple `noarch: python` packages

Meanwhile: removing `pip check` is not great, as this shifts the burden onto downstreams that _would_ like their stuff to be checkable... and most of the places I've hit this are indeed fairly complex.

Reply:
Once one starts patching dependencies to pass _some_ tests, I find it helpful to run _more_ tests, which are then basically testing the patch. In this case, any deviations in use of `typing_extensions.TypeGuard` and `typing_extensions.ParamSpec` probably _aren't_ going to get picked up by any existing test, as they might just be `import kombu`.

Reply:
I can't figure out how to run mypy correctly. Based on other recipies, it should be `mypy -p kombu`, but the distribution is missing the required `py.typed` markers. So I'm going to give up on patching the dependencies to remove the upper constraints for now, so that something can be uploaded.